File: 837-most-common-word/most-common-word.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        Set<String> ban = new HashSet<>(Arrays.asList(banned));

        Map<String, Integer> map = new HashMap<>();

        String ans = "";
        int max = 0;

        for (String w : paragraph.toLowerCase()
                .replaceAll("[^a-z]", " ")
                .split("\\s+")) {

            if (!ban.contains(w)) {

                int count = map.getOrDefault(w, 0) + 1;

                map.put(w, count);

                if (count > max) {
                    max = count;
                    ans = w;
                }
            }
        }

        return ans;

    }
}

File: 837-most-common-word/most-common-word.cpp
class Solution {
public:
    string mostCommonWord(string paragraph, vector<string>& banned) {

        unordered_set<string> ban(banned.begin(), banned.end());
        unordered_map<string,int> mp;

        string w, ans;
        int mx = 0;

        paragraph += ' ';

        for(char &c : paragraph) {

            if(isalpha(c)) {

                w += tolower(c);

            } else if(!w.empty()) {

                if(!ban.count(w)) {

                    if(++mp[w] > mx) {
                        mx = mp[w];
                        ans = w;
                    }
                }

                w.clear();
            }
        }

        return ans;
    }
};